tools/misc/tpm2_eventlog.c

```diff
@@ -27,10 +31,33 @@ static bool on_positional(int argc, char **argv) {
     return true;
 }
 
+static bool on_option(char key, char *value) {
+
+    int version;
+
+    switch (key) {
+    case 'y':
+        version = atoi(value);
+        for (unsigned int i = 0; i < sizeof(allowed_yaml_versions)/sizeof(int); i++) {
+            if (version == allowed_yaml_versions[i]) {
+                yaml_version = version;
+                return true;
+            }
+        }
+        LOG_ERR("Unexpected YAML version number: %d\n", version);
+        return false;
+    }
+    return true;
+}
+
 static bool tpm2_tool_onstart(tpm2_options **opts) {
 
-    *opts = tpm2_options_new(NULL, 0, NULL, NULL, on_positional,
-                             TPM2_OPTIONS_NO_SAPI);
+    static struct option topts[] = {
+         { "yaml-version",         required_argument, NULL, 'y' },
```

Comment:
@huang195 @williamcroberts not trying to be picky, but the option yaml-version is a bit of detour from what we used in specifying the versioning for context and session, there we referenced them as session_version and context_version.

Reply:
Also, maybe don't use a short option at all.

Reply:
> @huang195 @williamcroberts not trying to be picky, but the option yaml-version is a bit of detour from what we used in specifying the versioning for context and session, there we referenced them as session_version and context_version.

Also because, yaml outputs exist for other tools as well and things can get confusing.

Reply:
Sure, will rename `yaml-version` to `yaml_version`, and get rid of the short option.

Reply:
> Sure, will rename `yaml-version` to `yaml_version`, and get rid of the short option.

@huang195 sorry for not being specific, I meant, something like `eventlog-version` as yaml versioning has a connotation of being applied to rest of the tools as well. @huang195 @williamcroberts what do you think?

Reply:
@idesai ah gotcha. I agree - `eventlog-version` is more concise and would avoid the confusion. I've made the change